packages/flutter_tools/lib/src/commands/version.dart

```diff
@@ -57,8 +57,26 @@ class VersionCommand extends FlutterCommand {
     final List<String> tags = await getTags();
     if (argResults.rest.isEmpty) {
       tags.forEach(globals.printStatus);
-      return const FlutterCommandResult(ExitStatus.success);
+      return FlutterCommandResult.success();
     }
+
+    globals.printStatus(
+      'Warning: "flutter version" will leave the SDK in a detached HEAD state. '
```

Comment:
Consider adding an ascii border as in the welcome message in usage.dart.

Reply:
Done

Reply:
I verified the borders lined up in my own terminal, I think that should be sufficient unless someone isn't using a monospace font, right?

Reply:
If someone isn't using a monospace font in a terminal, then there's nothing we can do for them.